### PR TITLE
Revert "[MM-9563] Add a red dot to browser favicon

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Revert "[MM-9563] Add a red dot to browser favicon when there are unread mentions (#2926)"

This reverts commit dee85ecda1dac94e48964a060645e558a1a0397b.

This was previously reverted in the v5.14/v5.15 timeframe, but left in `master`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17618
https://mattermost.atlassian.net/browse/MM-17474

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/2926
https://github.com/mattermost/mattermost-webapp/commit/feec2fa787c14da02ad7b5ff01087b260892d397

[MM-9563]: https://mattermost.atlassian.net/browse/MM-9563

### CODE REVIEW

**Comment by lindalumitchell** [comment]
QA approves reverting this feature. Removing the QA Review label.

File: components/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -240,18 +237,26 @@ export default class Sidebar extends React.PureComponent {
...
+                link.href = typeof redFavicon === 'string' ? redFavicon : '';
+            } else {
+                link.href = typeof favicon === 'string' ? favicon : '';
+            }
+            var head = document.getElementsByTagName('head')[0];
+            var oldLink = document.getElementById('favicon');
+            if (oldLink) {
+                head.removeChild(oldLink);
+            }
+            head.appendChild(link);
```
Comment (mickmister) [inline components/sidebar/sidebar.jsx]: :+1:

File: components/sidebar/sidebar.jsx [hunk 2]
```diff
@@ -240,18 +237,26 @@ export default class Sidebar extends React.PureComponent {
...
     setBadgesActiveAndFavicon() {
-        const link = document.querySelector('link[rel="icon"]');
-
-        if (!link) {
-            return;
-        }
-
         this.lastBadgesActive = this.badgesActive;
-        this.badgesActive = this.props.unreads.mentionCount > 0;
+        this.badgesActive = this.props.unreads.mentionCount;
```
Comment (mickmister) [inline components/sidebar/sidebar.jsx]: `this.badgesActive` is set to `false` later on in this file. Do you think it makes sense to do something like this?
```suggestion
        this.badgesActive = Boolean(this.props.unreads.mentionCount);
```

**Comment by lieut-data** [comment]
Thanks for the careful review, @mickmister! This is just a vanilla revert of the PR in question, back to the original code as written. I'm hesitant to make further changes in a revert unless there's a merge conflict. Any objection to leaving the code as-is?

**Pull request merged**